packages/aws-cdk/toolkit/lib/cli-io-host.ts

```diff
@@ -0,0 +1,143 @@
+import * as chalk from 'chalk';
+
+/**
+ * Basic message structure for toolkit notifications.
+ * Messages are emitted by the toolkit and handled by the IoHost.
+ */
+interface IoMessage {
+  /**
+   * @param time The time the message was emitted.
+   */
+  readonly time: Date;
+
+  /**
+   * @param level The log level of the message.
+   */
+  readonly level: IoMessageLevel;
+
+  /**
+   * @param action The action that triggered the message.
+   */
+  readonly action: IoAction;
+
+  /**
+   * @param code A short code to identify the message.
```

Comment:
this needs more info. how will this be used? who needs to identify the message?

Reply:
This interface was defined in the RFC but what code represents wasn't explained. I pinged momo but we can hold off on this until he's back otherwise rather than guessing